### PR TITLE
UAR-784: Add statement validation middleware guards to update summary pages

### PR DESCRIPTION
**Pull request by conuirchgov**
### JIRA link

https://companieshouse.atlassian.net/browse/UAR-784

### Change description

Updated statement validation middleware to run on Check Your Answers and Review Update Statement pages. If validation fails on navigation to these pages, the user is redirected to the Secure Update Filter page (start of the journey).

### Work checklist

- [x] Tests added where applicable
- [x] UI changes meet accessibility criteria

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

**Review by cdrainCH — APPROVED** [review]

**Review by joshuahigginsonCH — APPROVED** [review]

File: src/middleware/statement.validation.middleware.ts [hunk 1]
```diff
@@ -4,67 +4,77 @@ import { checkActiveBOExists, checkActiveMOExists, getApplicationData, hasAddedO
...
-  const allOrSomeBOsIdentified: boolean = (appData[BeneficialOwnerStatementKey] === BeneficialOwnersStatementType.ALL_IDENTIFIED_ALL_DETAILS || appData[BeneficialOwnerStatementKey] === BeneficialOwnersStatementType.SOME_IDENTIFIED_ALL_DETAILS);
-  const someOrNoneBOsIdentified: boolean = (appData[BeneficialOwnerStatementKey] === BeneficialOwnersStatementType.SOME_IDENTIFIED_ALL_DETAILS || appData[BeneficialOwnerStatementKey] === BeneficialOwnersStatementType.NONE_IDENTIFIED);
-  const allBOsIdentified: boolean = (appData[BeneficialOwnerStatementKey] === BeneficialOwnersStatementType.ALL_IDENTIFIED_ALL_DETAILS);
-
-  if (allOrSomeBOsIdentified && !checkActiveBOExists(appData)) {
-    errorList.push(ErrorMessages.NO_ACTIVE_REGISTRABLE_BO);
+export const summaryPagesGuard = (req: Request, res: Response, next: NextFunction) => {
+  const hasStatementErrors = req['statementErrorList']?.length;
+  const flagEnabled = isActiveFeature(FEATURE_FLAG_ENABLE_UPDATE_STATEMENT_VALIDATION);
+  if (!flagEnabled || flagEnabled && !hasStatementErrors) {
```
Comment (AndreKramer7654) [inline src/middleware/statement.validation.middleware.ts:34]: you don't need second flagEnabled
 if (!flagEnabled || !hasStatementErrors) {

Comment (conuirchgov) [reply src/middleware/statement.validation.middleware.ts]: Indeed 🤦‍♂️. Removed 👍

File: src/controllers/update/update.statement.validation.errors.controller.ts [hunk 2]
```diff
@@ -38,7 +38,7 @@ export const post = (req: Request, res: Response, next: NextFunction) => {
     const inNoChangeJourney = !!appData.update?.no_change;
 
     if (!errors.isEmpty()) {
-      return renderPage(res, appData, inNoChangeJourney, req.body['statementErrorList[]'], formatValidationError(errors.array()));
+      return renderPage(res, appData, inNoChangeJourney, req['statementErrorList'], formatValidationError(errors.array()));
```
Comment (conuirchgov) [inline src/controllers/update/update.statement.validation.errors.controller.ts:41]: test

**Review by AndreKramer7654 — APPROVED** [review]
Conuir confirmed the redirect question so all good.

**Pull request merged**